engine/crates/engine-v2/src/sources/graphql/request/prepare.rs

```diff
@@ -164,6 +224,7 @@ impl QueryBuilderContext {
 
     fn write_selection_set(
         &mut self,
+        maybe_selection_set_type: Option<SelectionSetType>,
```

Comment:
You can remove the `maybe_entity_id` it's the same thing as `maybe_selection_set_type`, only makes the signature heavier than necessary and it's confusing as I'm not even sure you're providing the same thing for both. IMHO